tests/manager/manager-ctor-auth_mechanism-001.phpt

```diff
@@ -7,7 +7,7 @@ $tests = [
     ['mongodb://username@127.0.0.1/?authMechanism=MONGODB-X509', []],
     ['mongodb://username@127.0.0.1/?authMechanism=GSSAPI', []],
     [null, ['authMechanism' => 'MONGODB-X509']],
-    [null, ['authMechanism' => 'GSSAPI']],
+    [null, ['authMechanism' => 'GSSAPI', 'username' => 'username']],
```

Comment:
Quoting from the comment in the function itself:
> MONGODB-X509 is the only mechanism that doesn't require username

It may be confusing because the first test case passes a username, which could be removed as well. Either way, the test shouldn't fail regardless of whether a username was passed for X509 authentication.

Reply:
Would it make sense to include the extra two variations for X509 then?

```
['mongodb://127.0.0.1/?authMechanism=MONGODB-X509', []],
[null, ['authMechanism' => 'MONGODB-X509', 'username' => 'username']],
```

That would make for a good inline reminder that the username is optional if/when we come back to this test a few years from now.

Reply:
Absolutely. I've added these two test cases and also adapted the GSSAPI test above to include the same options as its connection string variant 👍